src/api/hooks.ts

```diff
@@ -252,3 +288,32 @@ export const useHasVisibleChildren = (propertyOwnerUri: Uri): boolean => {
 export function useIsConnectionStatus(status: ConnectionStatus): boolean {
   return useAppSelector((state) => state.connection.connectionStatus) === status;
 }
+
+export function useSetOpenSpaceTime() {
+  const luaApi = useOpenSpaceApi();
+
+  const setTime = (newTime: Date) => {
+    const fixedTimeString = dateToOpenSpaceTimeString(newTime);
+    luaApi?.time.setTime(fixedTimeString);
+  };
+
+  const interpolateTime = (newTime: Date) => {
+    const fixedTimeString = dateToOpenSpaceTimeString(newTime);
+    luaApi?.time.interpolateTime(fixedTimeString);
+  };
+
+  return { setTime, interpolateTime };
+}
+
+/**
+ * Hook that listens to a prop and updates the local state when the prop changes.
+ */
+export function usePropListeningState<T>(prop: T) {
```

Comment:
Naming this is hard... i kind of like something with cache in it? `useLocalCache`? `useLocalReduxCache`? This one doesn't really need to be a prop.